src/components/VisualizationPlugin/VisualizationPlugin.js

```diff
@@ -145,6 +216,30 @@ export const VisualizationPlugin = ({
             })
 
             if (result.responses.length) {
+                try {
+                    ensureAnalyticsResponsesContainData(
+                        result.responses,
+                        visualization.type
+                    )
+
+                    result.responses.forEach((response) => {
+                        if (
+                            (response?.metaData?.dimensions || {})[
+                                DIMENSION_ID_DATA
+                            ]?.every(
+                                (dim) =>
+                                    response.metaData.items[dim]?.valueType ===
```

Comment:
On line 227 you check defined for both `response` and `metadata`. Any need to do that here as well?